kuma/wiki/tests/test_views.py

```diff
@@ -3425,9 +3426,11 @@ def test_document_urls(self):
             d.locale = doc['kuma_locale']
             d.save()
             mt_url = '/%s' % '/'.join([doc['mt_locale'], doc['title']])
-            resp = self.client.get(mt_url)
-            eq_(301, resp.status_code)
-            eq_('http://testserver%s' % doc['expected'], resp['Location'])
+            resp = self.client.get(mt_url, follow=True)
+            eq_(200, resp.status_code)
+
+            # Check the last redirect chain url is the expected url
+            eq_('http://testserver%s' % doc['expected'], resp.redirect_chain[-1][0])
```

Comment:
I don't think the test should be altered in this way. After the changes, the request doesn't make it to the ``mindtouch_to_kuma_redirect``, but instead the ``LocaleURLMiddleware`` handles it.   I think this test and the supporting ``self.documents`` should be deleted.

If you want, you can add new test cases to ``LocalURLMiddleware``'s tests, in ``kuma/core/tests/test_locale_middleware.py``, which only tests the ``/en/`` case now.

https://github.com/mozilla/kuma/blob/c6c6bb0ae8493d71fb77dd1d1948f2c051394720/kuma/core/tests/test_locale_middleware.py#L37-L40

Reply:
@jwhitlock The request actually make it to `mindtouch_to_kuma_redirect`.
So it happens like this
`/cn/Web` redirect with 302 to `zh-cn/Web`
`zh-cn/Web` does not find any view, so it goes to `mindtouch_to_kuma_redirect` and redirect to `zh-cn/docs/Web`. So I think this test should be there